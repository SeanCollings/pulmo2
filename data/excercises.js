import { DATA, STRENGTH_KEY, ENDURANCE_KEY, CUSTOM_KEY } from '.';

const STRENGTH = {
  id: STRENGTH_KEY,
  title: STRENGTH_KEY,
  icon: 'dumbbell',
  contents: DATA[STRENGTH_KEY],
};
const ENDURANCE = {
  id: ENDURANCE_KEY,
  title: ENDURANCE_KEY,
  icon: 'clock',
  contents: DATA[ENDURANCE_KEY],
};
const CUSTOM = {
  id: CUSTOM_KEY,
  title: CUSTOM_KEY,
  icon: 'wrench',
  contents: DATA[CUSTOM_KEY],
};

export const EXCERCISE_DATA = [STRENGTH, ENDURANCE, CUSTOM];
