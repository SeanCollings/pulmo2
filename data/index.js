export const STRENGTH_KEY = 'Strength';
export const ENDURANCE_KEY = 'Endurance';
export const CUSTOM_KEY = 'Custom';

export const DATA = {
  [STRENGTH_KEY]: [
    {
      id: 11,
      title: 'Strength Level I',
      cycles: 10,
      rest: 60,
      rounds: 4,
    },
    {
      id: 12,
      title: 'Strength Level II',
      cycles: 10,
      rest: 60,
      rounds: 5,
    },
    {
      id: 13,
      title: 'Strength Level III',
      cycles: 12,
      rest: 50,
      rounds: 5,
    },
  ],
  [ENDURANCE_KEY]: [
    {
      id: 21,
      title: 'Endurance Level I',
      cycles: 10,
      rest: 30,
      rounds: 4,
    },
    {
      id: 22,
      title: 'Endurance Level II',
      cycles: 12,
      rest: 20,
      rounds: 5,
    },
    {
      id: 23,
      title: 'Endurance Level III',
      cycles: 15,
      rest: 20,
      rounds: 5,
    },
  ],
  [CUSTOM_KEY]: [],
};

export const DEFAULT_EXCERCISE = DATA[STRENGTH_KEY][0];
