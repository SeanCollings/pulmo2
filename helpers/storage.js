import { AsyncStorage } from 'react-native';
import { APP_ID } from '../constants/constants';

export const clearAllAsyncStorage = () => {
  AsyncStorage.clear();
};

export const getAsyncData = async (id) => {
  try {
    const value = await AsyncStorage.getItem(`${APP_ID}${id}`);
    if (value !== null) {
      return JSON.parse(value);
    } else {
      return null;
    }
  } catch (error) {
    console.log('Failure retrieving ' + id + ' error: ' + error);
  }
};

export const getMultiAsyncData = async (id, keys) => {
  try {
    const formattedKeys = keys.map((key) => `${APP_ID}${id}${key}`);
    const values = await AsyncStorage.multiGet(formattedKeys);

    if (values && !!values.length) return values;
    else {
      return null;
    }
  } catch (err) {
    console.log('Failure retrieving from all keys with error ' + error);
  }
};

export const storeAsyncData = async (id, value) => {
  try {
    await AsyncStorage.setItem(`${APP_ID}${id}`, JSON.stringify(value));
    return true;
  } catch (error) {
    console.log('Failure saving: ' + error);
    return false;
  }
};

export const removeAsyncData = async (id) => {
  try {
    await AsyncStorage.removeItem(`${APP_ID}${id}`);
  } catch (err) {
    console.log('Failure removing: ' + error);
  }
};
