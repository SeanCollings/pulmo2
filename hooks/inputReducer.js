import { INSTRUCTIONS } from '../constants/constants';

export const INPUT_TOGGLE = 'INPUT_TOGGLE';
export const INPUT_BREATH = 'INPUT_BREATH';
export const INPUT_REST = 'INPUT_REST';
export const INPUT_END = 'INPUT_END';
export const INPUT_STOP = 'INPUT_STOP';

export const inputInitialState = {
  isActive: false,
  currentRound: 0,
  instructions: INSTRUCTIONS.initial,
  countDownTime: 0,
};

export const inputReducer = (state, action) => {
  switch (action.type) {
    case INPUT_TOGGLE:
      return { ...state, isActive: action.isActive };
    case INPUT_BREATH:
      return {
        ...state,
        isActive: false,
        currentRound: action.currentRound,
        instructions: INSTRUCTIONS.breathing,
        countDownTime: action.countDownTime,
      };
    case INPUT_REST:
      return {
        ...state,
        isActive: true,
        instructions: INSTRUCTIONS.rest,
      };
    case INPUT_END:
      return {
        ...state,
        isActive: false,
        instructions: INSTRUCTIONS.end,
        countDownTime: 0,
      };
    case INPUT_STOP:
      return inputInitialState;
    default:
      return state;
  }
};
