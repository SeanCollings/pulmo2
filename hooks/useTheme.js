import { useContext } from 'react';

import { THEME, SettingsContext } from '../context/settings-context';
import { DARK_MODE, COLORS_DARK, COLORS } from '../constants/constants';

export const useTheme = () => {
  const { settingsContext } = useContext(SettingsContext);

  return settingsContext[THEME] === DARK_MODE ? COLORS_DARK : COLORS;
};
