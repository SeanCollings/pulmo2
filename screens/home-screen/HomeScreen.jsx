import React, {
  useCallback,
  useReducer,
  useState,
  useContext,
  useEffect,
} from 'react';
import { StyleSheet, View, Text, ScrollView, AppState } from 'react-native';
import { useIsFocused } from '@react-navigation/native';
import { useKeepAwake } from 'expo-keep-awake';
import * as SplashScreen from 'expo-splash-screen';

import { ExcerciseContext } from '../../context/excercise-context';
import { CustomExcerciseContext } from '../../context/custom-excercise-context';
import { ProfileContext, SELECTED_LEVEL } from '../../context/profile-context';
import { HistoryContext } from '../../context/history-context';
import options from './options';
import Timer from '../../components/Timer';
import {
  CircleIconButton,
  CircleTextButton,
} from '../../components/CircleButton';
import {
  INTIAL_STATE,
  BREATHING_STATE,
  END_STATE,
  REST_STATE,
} from '../../constants/constants';
import {
  inputInitialState,
  inputReducer,
  INPUT_BREATH,
  INPUT_END,
  INPUT_REST,
  INPUT_TOGGLE,
  INPUT_STOP,
} from '../../hooks/inputReducer';
import Table from '../../components/Table';
import ModalBegin from '../../components/modals/ModalBegin';
import {
  formatMilliToSeconds,
  getRemainingTime,
  getExcerciseById,
} from '../../utils';
import { useTheme } from '../../hooks/useTheme';
import AnimatedUnderline from '../../components/animated/AnimatedUnderline';
import AnimatedCycleText from '../../components/animated/AnimatedCycleText';
import ModalEndActivity from '../../components/modals/ModalEndActivity';
import Counter from '../../components/Counter';

export const homeScreenOptions = options;

let timeStampBreath;
let timeStampRest;

const HomeScreen = ({ navigation }) => {
  useKeepAwake();
  const theme = useTheme();
  const isFocused = useIsFocused();
  const { selectedExcercise } = useContext(ExcerciseContext);
  const { customExcercises } = useContext(CustomExcerciseContext);
  const { profileContext, updateWorkAverageDeviation } = useContext(
    ProfileContext
  );
  const { addActivity } = useContext(HistoryContext);

  const [showModal, setShowModal] = useState(false);
  const [showEndActivityModal, setShowEndActivityModal] = useState(false);
  const [userTimes, setUserTimes] = useState([]);
  const [currentBreathTime, setCurrentBreathTime] = useState('');
  const [_, setAppState] = useState(AppState.currentState);

  const [
    { isActive, currentRound, instructions, countDownTime },
    dispatch,
  ] = useReducer(inputReducer, inputInitialState);

  const activeExcercise = selectedExcercise
    ? getExcerciseById(
        selectedExcercise[0],
        selectedExcercise[1],
        customExcercises
      )
    : {};

  const handleAppStateChange = (nextAppState) => {
    if (nextAppState === 'background') {
      dispatch({ type: INPUT_TOGGLE, isActive: false });
    }
    setAppState(nextAppState);
  };

  useEffect(() => {
    SplashScreen.hideAsync();
    AppState.addEventListener('change', handleAppStateChange);
    return () => AppState.removeEventListener('change', handleAppStateChange);
  }, []);

  useEffect(() => {
    if (!isFocused && instructions.state !== INTIAL_STATE) {
      dispatch({ type: INPUT_STOP });
      setUserTimes([]);
      setCurrentBreathTime('');
    }
    return () => {};
  }, [isFocused, instructions.state]);

  useEffect(() => {
    if (selectedExcercise) {
      dispatch({ type: INPUT_STOP });
      setUserTimes([]);
    }
  }, [selectedExcercise]);

  const toggleHandler = () => {
    if (instructions.state === END_STATE) {
      setUserTimes([]);
      dispatch({ type: INPUT_STOP });
    } else if (currentRound > 0) {
      dispatch({ type: INPUT_TOGGLE, isActive: !isActive });
    } else {
      setShowModal(true);
    }
  };

  const timerFinished = useCallback(
    (timeNow) => {
      if (currentRound < +activeExcercise.rounds) {
        const timeDifferenceRest = timeNow - timeStampRest;
        const restPausedExtra =
          Math.floor(timeDifferenceRest / 1000) - countDownTime;
        const totalCountDown =
          restPausedExtra > 15 ? restPausedExtra : countDownTime;

        setUserTimes((curr) => [
          ...curr,
          [currentBreathTime, getRemainingTime(totalCountDown)],
        ]);

        timeStampBreath = Date.now();
        setCurrentBreathTime(timeStampBreath);

        dispatch({
          type: INPUT_BREATH,
          currentRound: currentRound + 1,
          countDownTime: activeExcercise.rest,
        });
      }
    },
    [currentRound, currentBreathTime, timeStampRest]
  );

  const endCurrentActivity = () => {
    dispatch({ type: INPUT_STOP });
    setUserTimes([]);
    setCurrentBreathTime(null);
  };

  const stopTimerHandler = () => {
    if (!!userTimes.length) {
      dispatch({ type: INPUT_TOGGLE, isActive: false });
      setShowEndActivityModal(true);
    } else {
      endCurrentActivity();
    }
  };

  const saveActivity = (timeDifferenceBreath, incomplete) => {
    const timeBreath = timeDifferenceBreath ? [timeDifferenceBreath, ''] : [];
    const results = [...userTimes, timeBreath];

    updateWorkAverageDeviation(results);
    addActivity({
      date: new Date().toISOString(),
      excercise: activeExcercise,
      level: profileContext[SELECTED_LEVEL],
      results,
      type: selectedExcercise[1],
      incomplete,
    });
  };

  const toggleRestHandler = useCallback(() => {
    const dateNow = Date.now();
    const timeDifferenceBreath = formatMilliToSeconds(
      dateNow - timeStampBreath
    );

    if (currentRound === +activeExcercise.rounds) {
      saveActivity(timeDifferenceBreath);

      setUserTimes((curr) => [...curr, [timeDifferenceBreath, '']]);
      dispatch({ type: INPUT_END });
    } else {
      setCurrentBreathTime(timeDifferenceBreath);
      dispatch({ type: INPUT_REST });
      timeStampRest = Date.now();
    }
  }, [currentRound, currentBreathTime, timeStampBreath]);

  const closeModalHandler = (startWorkout) => {
    setShowModal(false);

    if (startWorkout) {
      timeStampBreath = Date.now();
      dispatch({
        type: INPUT_BREATH,
        currentRound: 1,
        countDownTime: activeExcercise.rest,
      });
    }
  };

  const resumeActivityHandler = () => {
    setShowEndActivityModal(false);

    if (instructions.state === REST_STATE)
      dispatch({ type: INPUT_TOGGLE, isActive: true });
  };
  const cancelEndActivityModalHandler = () => {
    setShowEndActivityModal(false);
    endCurrentActivity();
  };
  const confirmEndActivityModalHandler = (reason) => {
    saveActivity(
      instructions.state === BREATHING_STATE ? null : currentBreathTime,
      [reason]
    );
    setShowEndActivityModal(false);
    endCurrentActivity();
  };

  if (!activeExcercise) return null;

  let progressText = `selected level: ${profileContext[SELECTED_LEVEL]}`;
  if (currentRound > 0 && instructions.state !== END_STATE) {
    progressText = `round ${currentRound}/${activeExcercise.rounds}`;
  } else if (instructions.state === END_STATE) {
    progressText = 'restart';
  }

  let buttonIconName = 'play';
  if (isActive) buttonIconName = 'pause';
  else if (instructions.state === END_STATE) buttonIconName = 'restart';

  const breathingState = instructions.state === BREATHING_STATE;
  const showRestartButton =
    instructions.state !== INTIAL_STATE && instructions.state !== END_STATE;

  const opacity = theme.DARK ? 0.87 : 1;
  const styleSubText = {
    ...styles.subText,
    color: theme.TEXT,
    opacity,
  };

  return (
    <View style={{ flex: 1 }}>
      <ScrollView contentContainerStyle={{ flexGrow: 1 }}>
        <View
          style={{
            ...styles.container,
            backgroundColor: theme.BACKGROUND,
          }}
        >
          <View style={styles.timerContainer}>
            <View
              style={{
                ...(!breathingState ? {} : { height: 0, opacity: 0 }),
              }}
            >
              <Timer
                isActive={isActive}
                timerFinished={(timeNow) => timerFinished(timeNow)}
                replacementText={!isActive && instructions.prompt}
                countDownTime={countDownTime}
              />
            </View>
            {breathingState && (
              <AnimatedUnderline>
                <AnimatedCycleText
                  textArray={['Breathe', instructions.prompt]}
                />
              </AnimatedUnderline>
            )}
          </View>
          <View style={styles.buttonContainer}>
            {!breathingState && (
              <CircleIconButton
                onPress={toggleHandler}
                name={buttonIconName}
                borderWidth={4}
                size={60}
                buttonSize={120}
                iconStyle={{ color: theme.SECONDARY }}
                style={{ borderColor: theme.SECONDARY }}
              />
            )}
            {breathingState && (
              <CircleTextButton
                onPress={toggleRestHandler}
                borderWidth={4}
                buttonSize={120}
                fontSize={30}
                text={
                  currentRound !== +activeExcercise.rounds ? 'REST' : 'DONE'
                }
                textStyle={{ color: theme.SECONDARY }}
                style={{ borderColor: theme.SECONDARY }}
              />
            )}
            {showRestartButton && (
              <CircleIconButton
                onPress={stopTimerHandler}
                name={'restart'}
                iconSize={40}
                buttonSize={56}
                borderWidth={0}
                style={styles.restartButtonContainer}
                iconStyle={{ color: theme.TERTIARY }}
              />
            )}
          </View>
          <Text style={{ ...styles.progressText, color: theme.TEXT, opacity }}>
            {progressText}
          </Text>
          <View style={styles.instructionContainer}>
            <Text style={{ ...styles.ExcerciseTitle, color: theme.TERTIARY }}>
              {activeExcercise.title}
            </Text>
            {instructions.state !== END_STATE &&
              instructions.state !== REST_STATE && (
                <View style={styles.gridContainer}>
                  <Text
                    style={{
                      ...styleSubText,
                      ...styles.textAlignRight,
                      borderRightColor: theme.BORDER,
                    }}
                  >{`inhale & exhale`}</Text>
                  <Text
                    style={styleSubText}
                  >{`${activeExcercise.cycles} times`}</Text>
                </View>
              )}
            {(instructions.state === INTIAL_STATE ||
              instructions.state === REST_STATE) && (
              <View style={styles.gridContainer}>
                <Text
                  style={{
                    ...styleSubText,
                    ...styles.textAlignRight,
                    borderRightColor: theme.BORDER,
                  }}
                >{`rest for`}</Text>
                <Text
                  style={styleSubText}
                >{`${activeExcercise.rest} seconds`}</Text>
              </View>
            )}
            {instructions.state === INTIAL_STATE && (
              <View style={styles.gridContainer}>
                <Text
                  style={{
                    ...styleSubText,
                    ...styles.textAlignRight,
                    borderRightColor: theme.BORDER,
                  }}
                >{`rounds`}</Text>
                <Text style={styleSubText}>{`${activeExcercise.rounds}`}</Text>
              </View>
            )}
          </View>
          <View style={styles.tableContainer}>
            <Table
              headerContent={['Workout', 'Rest']}
              rowContents={userTimes}
              headerColour={theme.DARK ? theme.TEXT : theme.TERTIARY}
            />
          </View>
          {showModal && (
            <ModalBegin
              cancelModal={() => closeModalHandler(false)}
              confirmModal={() => closeModalHandler(true)}
              navigation={navigation}
            />
          )}
          {showEndActivityModal && (
            <ModalEndActivity
              cancelModal={cancelEndActivityModalHandler}
              confirmModal={confirmEndActivityModalHandler}
              resumeActivity={resumeActivityHandler}
            />
          )}
        </View>
      </ScrollView>
      {showRestartButton && (
        <Counter
          disabled={instructions.state === REST_STATE}
          reset={instructions.state === BREATHING_STATE}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    paddingBottom: 70,
  },
  timerContainer: {
    height: 100,
    justifyContent: 'center',
  },
  buttonContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
  },
  instructionContainer: {
    paddingTop: 20,
    paddingBottom: 10,
    justifyContent: 'center',
    alignItems: 'center',
    width: '90%',
  },
  tableContainer: { marginTop: 20, width: '100%', alignItems: 'center' },
  progressText: {
    fontSize: 20,
    fontFamily: 'tit-light',
    marginTop: 10,
  },
  ExcerciseTitle: {
    fontSize: 22,
    fontFamily: 'tit-regular',
    paddingBottom: 5,
  },
  gridContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  subText: {
    fontSize: 17,
    paddingTop: 5,
    fontFamily: 'tit-light',
    width: '50%',
    paddingHorizontal: 10,
  },
  textAlignRight: {
    textAlign: 'right',
    borderRightWidth: 1,
  },
  restartButtonContainer: {
    position: 'absolute',
    right: '15%',
    top: 0,
  },
});

export default HomeScreen;
