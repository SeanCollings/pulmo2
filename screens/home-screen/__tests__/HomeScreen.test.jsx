import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup, fireEvent } from 'react-native-testing-library';
import HomeScreen, { homeScreenOptions } from '../HomeScreen';
import HistoryContextProvider from '../../../context/history-context';
import CustomExcerciseContext from '../../../context/custom-excercise-context';
import ProfileContext from '../../../context/profile-context';
import ExcerciseContext from '../../../context/excercise-context';
import { useIsFocused } from '@react-navigation/native';
import { getAsyncData } from '../../../helpers/storage';
import { ENDURANCE_KEY } from '../../../data';
import { INPUT_END } from '../../../hooks/inputReducer';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.mock('@react-navigation/native');
jest.mock('../../../helpers/storage');

describe('HomeScreen - unit test', () => {
  let component;
  const navigation = { goBack: () => {} };

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test('should render the default excercise if there is no excercise', async () => {
    useIsFocused.mockReturnValueOnce(true);
    getAsyncData.mockReturnValueOnce(null);

    await act(async () => {
      component = create(
        <HistoryContextProvider>
          <CustomExcerciseContext>
            <ProfileContext>
              <ExcerciseContext>
                <HomeScreen navigation={navigation} />
              </ExcerciseContext>
            </ProfileContext>
          </CustomExcerciseContext>
        </HistoryContextProvider>
      );
    });

    expect(component).toMatchSnapshot();
  });

  test('should render the active excercise if there is one', async () => {
    useIsFocused.mockReturnValueOnce(true);
    getAsyncData.mockReturnValueOnce([21, ENDURANCE_KEY]);

    await act(async () => {
      component = create(
        <HistoryContextProvider>
          <CustomExcerciseContext>
            <ProfileContext>
              <ExcerciseContext>
                <HomeScreen navigation={navigation} />
              </ExcerciseContext>
            </ProfileContext>
          </CustomExcerciseContext>
        </HistoryContextProvider>
      );
    });

    expect(component).toMatchSnapshot();
  });

  describe('homeScreenOptions', () => {
    test('options', () => {
      const options = homeScreenOptions();
      expect(options).toMatchSnapshot();
    });

    test('headerTitle', () => {
      const headerTitle = homeScreenOptions().headerTitle({
        tintColor: 'white',
      });
      expect(headerTitle).toMatchSnapshot();
    });

    test('headerRight', () => {
      const headerRight = homeScreenOptions().headerRight({
        tintColor: 'white',
      });
      expect(headerRight).toMatchSnapshot();
    });

    test('should handle headerRight navigate to HomeHelp', async () => {
      const navigate = jest.fn();
      const navData = { navigation: { navigate } };

      await act(async () => {
        component = create(
          homeScreenOptions(navData).headerRight({
            tintColor: 'white',
          })
        );
      });

      const headerButton = component.root.find(
        ({ props }) => props.title === 'help'
      );
      fireEvent.press(headerButton);
      expect(navigate).toHaveBeenCalledWith('HomeHelp');
    });

    test('tabBarIcon', () => {
      const tabBarIcon = homeScreenOptions().tabBarIcon({
        color: 'white',
      });
      expect(tabBarIcon).toMatchSnapshot();
    });
  });
});
