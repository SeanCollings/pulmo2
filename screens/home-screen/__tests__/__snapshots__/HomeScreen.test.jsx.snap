// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen - unit test homeScreenOptions headerRight 1`] = `
<HeaderButtons
  HeaderButtonComponent={[Function]}
  left={false}
>
  <Item
    color="white"
    iconName="help-circle-outline"
    onPress={[Function]}
    title="help"
  />
</HeaderButtons>
`;

exports[`HomeScreen - unit test homeScreenOptions headerTitle 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
      }
    }
  >
    PulmO
  </Text>
  <View
    style={
      Object {
        "top": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
        }
      }
    >
      2
    </Text>
  </View>
</View>
`;

exports[`HomeScreen - unit test homeScreenOptions options 1`] = `
Object {
  "headerRight": [Function],
  "headerTitle": [Function],
  "tabBarIcon": [Function],
}
`;

exports[`HomeScreen - unit test homeScreenOptions tabBarIcon 1`] = `
<Icon
  allowFontScaling={false}
  color="white"
  name="home"
  size={26}
/>
`;

exports[`HomeScreen - unit test should render the active excercise if there is one 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "paddingBottom": 70,
          }
        }
      >
        <View
          style={
            Object {
              "height": 100,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={Object {}}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3e4b57",
                    "fontFamily": "tit-light",
                    "fontSize": 40,
                    "height": 60,
                    "opacity": 1,
                    "textAlignVertical": "center",
                    "textTransform": "lowercase",
                  }
                }
              >
                tap to begin
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#002f56",
                "borderRadius": 120,
                "borderWidth": 4,
                "height": 120,
                "justifyContent": "center",
                "opacity": 1,
                "width": 120,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 60,
                  },
                  Object {
                    "color": "#002f56",
                  },
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              Ôêä
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-light",
              "fontSize": 20,
              "marginTop": 10,
              "opacity": 1,
            }
          }
        >
          selected level: 5
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingTop": 20,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#00b8bb",
                "fontFamily": "tit-regular",
                "fontSize": 22,
                "paddingBottom": 5,
              }
            }
          >
            Endurance Level I
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <Text
              style={
                Object {
                  "borderRightColor": "#CCC",
                  "borderRightWidth": 1,
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "textAlign": "right",
                  "width": "50%",
                }
              }
            >
              inhale & exhale
            </Text>
            <Text
              style={
                Object {
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "width": "50%",
                }
              }
            >
              10 times
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <Text
              style={
                Object {
                  "borderRightColor": "#CCC",
                  "borderRightWidth": 1,
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "textAlign": "right",
                  "width": "50%",
                }
              }
            >
              rest for
            </Text>
            <Text
              style={
                Object {
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "width": "50%",
                }
              }
            >
              30 seconds
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <Text
              style={
                Object {
                  "borderRightColor": "#CCC",
                  "borderRightWidth": 1,
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "textAlign": "right",
                  "width": "50%",
                }
              }
            >
              rounds
            </Text>
            <Text
              style={
                Object {
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "width": "50%",
                }
              }
            >
              4
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginTop": 20,
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`HomeScreen - unit test should render the default excercise if there is no excercise 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flex": 1,
            "paddingBottom": 70,
          }
        }
      >
        <View
          style={
            Object {
              "height": 100,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={Object {}}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#3e4b57",
                    "fontFamily": "tit-light",
                    "fontSize": 40,
                    "height": 60,
                    "opacity": 1,
                    "textAlignVertical": "center",
                    "textTransform": "lowercase",
                  }
                }
              >
                tap to begin
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#002f56",
                "borderRadius": 120,
                "borderWidth": 4,
                "height": 120,
                "justifyContent": "center",
                "opacity": 1,
                "width": 120,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 60,
                  },
                  Object {
                    "color": "#002f56",
                  },
                  Object {
                    "fontFamily": "material-community",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              Ôêä
            </Text>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#3e4b57",
              "fontFamily": "tit-light",
              "fontSize": 20,
              "marginTop": 10,
              "opacity": 1,
            }
          }
        >
          selected level: 5
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingTop": 20,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#00b8bb",
                "fontFamily": "tit-regular",
                "fontSize": 22,
                "paddingBottom": 5,
              }
            }
          >
            Strength Level I
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <Text
              style={
                Object {
                  "borderRightColor": "#CCC",
                  "borderRightWidth": 1,
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "textAlign": "right",
                  "width": "50%",
                }
              }
            >
              inhale & exhale
            </Text>
            <Text
              style={
                Object {
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "width": "50%",
                }
              }
            >
              10 times
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <Text
              style={
                Object {
                  "borderRightColor": "#CCC",
                  "borderRightWidth": 1,
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "textAlign": "right",
                  "width": "50%",
                }
              }
            >
              rest for
            </Text>
            <Text
              style={
                Object {
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "width": "50%",
                }
              }
            >
              60 seconds
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
              }
            }
          >
            <Text
              style={
                Object {
                  "borderRightColor": "#CCC",
                  "borderRightWidth": 1,
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "textAlign": "right",
                  "width": "50%",
                }
              }
            >
              rounds
            </Text>
            <Text
              style={
                Object {
                  "color": "#3e4b57",
                  "fontFamily": "tit-light",
                  "fontSize": 17,
                  "opacity": 1,
                  "paddingHorizontal": 10,
                  "paddingTop": 5,
                  "width": "50%",
                }
              }
            >
              4
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "marginTop": 20,
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
